.github/workflows/tests.yml

```diff
@@ -13,51 +13,56 @@ concurrency:
 permissions:
   contents: read
 
+env:
+  DJANGO_SETTINGS_MODULE: tracdjangoplugin.settings_tests
+  PYTHONWARNINGS: >
+    error,
+    default:::passlib.utils,
+
 jobs:
   noshadows:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-24.04
     steps:
       - name: Checkout
         uses: actions/checkout@v4
       - uses: actions/setup-python@v5
         with:
-          python-version: '3.8'
+          python-version: '3.11'
       - run: pip install "tinycss2>=1.2.0"
       - run: python noshadows.py --tests
-        env:
-          PYTHONWARNINGS: error
 
   tracdjangoplugin:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-24.04
     steps:
       - name: Checkout
         uses: actions/checkout@v4
       - uses: actions/setup-python@v5
         with:
-          python-version: '3.8'
+          python-version: '3.11'
+      - name: Install system package dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get -y install subversion
```

Comment:
I'm guessing this is because subversion isn't included in ubuntu 24?

Reply:
Yes, exactly, and we have two dependencies that still use SVN 🙀